lib/Runtime/Base/FunctionBody.h

```diff
@@ -316,13 +316,28 @@ namespace Js
     public:
         // These are public because we don't manage them nor their consistency;
         // the user of this class does.
-        void * address;
+        union address_type_ {
+            address_type_(void *ptr):asPtr(ptr) {}
+            address_type_(Js::JavascriptMethod method):asJsMethod(method) {}
 
-        ProxyEntryPointInfo(void* address, ThreadContext* context = nullptr):
+            void *asPtr;
```

Comment:
> writing one member of a union and then inspecting it through another member leads to undefined behavior in C++ 

Right. See also (https://isocpp.org/wiki/faq/cpp11-language-types#generalized-unions) for a better reference. AFAIK, it is undefined due to different/unknown alignment approaches. 

However, it does not apply here. There is no casting from dynamically allocated memory to union / fun ptr.